app/main.py

```diff
@@ -1,2 +1,52 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+
+    def resize(self) -> None:
+        new_hash_table = [None] * len(self.hash_table) * 2
+        for i in range(len(self.hash_table)):
+            if self.hash_table[i] is not None:
+                key, hash_key, value = self.hash_table[i]
+                index = hash_key % len(new_hash_table)
+                while True:
+                    if new_hash_table[index] is None:
+                        new_hash_table[index] = self.hash_table[i]
+                        break
+                    if key == new_hash_table[index][0]:
+                        new_hash_table[index][2] = value
+                        break
+                    index = (index + 1) % len(new_hash_table)
+        self.hash_table = new_hash_table
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        hash_key = hash(key)
+        if self.length > len(self.hash_table) * 2 / 3:
+            self.resize()
+        index = hash_key % len(self.hash_table)
```

Comment:
index should be calculated like: `hash_key % capacity`
You can store capacity separately in different instance attribute)